src/ServerRequestFactoryInterface.php

```diff
@@ -12,20 +12,9 @@
      *
      * @param string $method
      * @param UriInterface|string $uri
+     * @param array $server optional server params ($_SERVER or similar structure)
      *
      * @return ServerRequestInterface
      */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from server variables.
-     *
-     * @param array $server Typically $_SERVER or similar structure.
-     *
-     * @return ServerRequestInterface
-     *
-     * @throws \InvalidArgumentException
-     *  If no valid method or URI can be determined.
-     */
-    public function createServerRequestFromArray(array $server);
+    public function createServerRequest($method, $uri, $server = []);
```

Comment:
@mindplay-dk ah, this is the root of the misunderstanding. The `$server` is not just "server params" but the entire `$_SERVER`, which contains enough data to construct the entire server request, including the URI, headers, etc.

Every PSR-7 implementation that I know of includes a method that uses (global) data to create a server request that represents that data. As such, I believe PSR-17 should include that functionality. What you are suggesting would confuse that significantly.